sdk/core/core-http/lib/util/xml.ts

```diff
@@ -34,4 +33,167 @@ export function parseXML(str: string): Promise<any> {
       });
     }
   });
+  return result;
+}
+
+export async function convertAtomXmlToJson(body: string): Promise<any> {
+  const parser = new xml2js.Parser(_getDefaultSettingsForAtomXmlOperations());
+  const result = await new Promise((resolve, reject) => {
+    parser.parseString(removeBOM(body.toString()), function(err: any, parsedBody: any) {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(parsedBody);
+      }
+    });
+  });
+  return result;
+}
+
+/**
+ * @ignore
+ * @param {object} content The content payload as it is to be serialized. It should include any root node(s).
+ */
+export function convertJsonToAtomXml(content: any): string {
+  content[Constants.XML_METADATA_MARKER] = { type: "application/xml" };
+
+  let doc = xmlbuilder.create();
+
+  doc = doc
+    .begin("entry", { version: "1.0", encoding: "utf-8", standalone: "yes" })
+    .att("xmlns", "http://www.w3.org/2005/Atom");
+
+  doc = doc.ele("updated", new Date().toISOString()).up();
+
+  doc = writeElementValue(doc, "content", content);
+  return doc.doc().toString();
+}
+
+/**
+ * @ignore
+ * Gets the default xml2js settings applicable for Atom based XML operations.
+ */
+function _getDefaultSettingsForAtomXmlOperations(): any {
+  const xml2jsSettings = xml2js.defaults["0.2"];
```

Comment:
I think this is why you're seeing setting shared across instances of the xml parser.
You are directly referencing the default settings and mutating them in this function.

Instead, you should create a copy of the default settings, then mutate the copy.

Reply:
Oh, good catch! This should be it - will test things and update.
Is this standard Javascript thing, where people remember to make copies? 
Because it seems like it's possible to make objects immutable - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze
Shall we still file an issue on xml2js repo?

Reply:
I think xml2js was written before Object.freeze existed, so it depends what runtimes they want to continue supporting, but feel free to log an issue.

However, I don't think the intent was for users to use `defaults["0.2"]`. It was exposed as a way to let users updating from version 0.1 to 0.2 test the new default settings before [updating](https://www.npmjs.com/package/xml2js#updating-to-new-version). I'm not sure there's any value in us using it directly since the default options will already be used for any fields we don't specify.

As far as what we can do if we want to be extremely careful and protect against another library making the same mistake we are, we can explicitly set all of our options, using https://github.com/Leonidas-from-XIV/node-xml2js/blob/aefc64af9e9badbd0aade00814548f2ea33c9b4e/lib/defaults.js#L29 as a starting point.

Reply:
Agree, addressed this along with the new set of changes being discussed @chradek @ramya-rao-a